felix/bpf/bpf_syscall.go

```diff
@@ -269,8 +269,16 @@ func checkMapIfDebug(mapFD MapFD, keySize, valueSize int) error {
 		if keySize != mapInfo.KeySize {
 			log.WithField("mapInfo", mapInfo).WithField("keyLen", keySize).Panic("Incorrect key length")
 		}
-		if valueSize >= 0 && valueSize != mapInfo.ValueSize {
-			log.WithField("mapInfo", mapInfo).WithField("valueLen", valueSize).Panic("Incorrect value length")
+		// The actual size of per cpu maps is equal to the value size * number of cpu
+		if mapInfo.Type == unix.BPF_MAP_TYPE_PERCPU_HASH || mapInfo.Type == unix.BPF_MAP_TYPE_PERCPU_ARRAY ||
```

Comment:
A suggestion, smaller, easier to add values in the future.

```suggestion
		switch (mapInfo.Type) {
		case unix.BPF_MAP_TYPE_PERCPU_HASH, unix.BPF_MAP_TYPE_PERCPU_ARRAY:
		    ...
		default:
		    ...
```